modules/server/src/main/scala/scaladex/server/service/ArtifactConverter.scala

```diff
@@ -63,6 +56,22 @@ class ArtifactConverter(paths: DataPaths) extends LazyLogging {
       (artifact, dependencies)
     }
 
+  private def extractScalaVersion(pom: ArtifactModel): Option[SemanticVersion] = {
+    val scalaDependencies = pom.dependencies.filter { dep =>
+      dep.groupId == "org.scala-lang" &&
+      (dep.artifactId == "scala-library" || dep.artifactId == "scala3-library_3")
+    }
+
+    val sortedScalaDependencies = scalaDependencies.sortBy(_.artifactId)
+    val fullScalaVersion = scalaDependencies match {
+      case _ :: _ :: _ => Some(sortedScalaDependencies.last.version)
+      case head :: Nil => Some(head.version)
+      case _           => None
+    }
```

Comment:
I think this is the same as:
```suggestion
    val fullScalaVersion = scalaDependencies.sortBy(_.artifactId).lastOption.map(_.version)
```